# Cross compile from linux for mingw32 don't init the ctors from crtend.o!
# Every 'throw' in a program crashes with a "NULL-Pointer read".
# 
--- binutils-2.15.91-20040904-1/ld/scripttempl/pe.sc.old
+++ binutils-2.15.91-20040904-1/ld/scripttempl/pe.sc
@@ -66,14 +66,14 @@
 			*(EXCLUDE_FILE (*crtend.o) .ctors);
 			*(.ctor);
 			*(SORT(.ctors.*));
-			*crtend.o (.ctors);
+			*(.ctors);
 			LONG (0); }
     ${CONSTRUCTING+	___DTOR_LIST__ = .; __DTOR_LIST__ = . ;
 			LONG (-1);
 			*(EXCLUDE_FILE (*crtend.o) .dtors);
 			*(.dtor);
 			*(SORT(.dtors.*));
-			*crtend.o (.dtors);
+			*(.dtors);
 			LONG (0); }
     ${RELOCATING+ *(.fini)}
     /* ??? Why is .gcc_exc here?  */
